src/block-tracker-inspector.ts

```diff
@@ -13,37 +16,64 @@ interface BlockTrackerInspectorMiddlewareOptions {
   blockTracker: PollingBlockTracker;
 }
 
+function hasProperty<Property extends string>(
+  object: unknown,
+  property: Property,
+): object is Record<Property, unknown> {
+  return Object.hasOwnProperty.call(object, property);
+}
+
+function getResultBlockNumber(
```

Comment:
The type narrowing here was painful to work through. Not sure if there's an easier way to do this